scripts/ci.sh

```diff
@@ -85,7 +86,7 @@ setup() {
 
 run_test() {
     echo "Running lint..."
-    golangci-lint run --no-config --disable-all --enable gosimple --enable govet --enable unused --enable deadcode --timeout 3m
+    golangci-lint run --no-config --disable-all --enable gosimple --enable errcheck --enable govet --enable unused --enable goconst --timeout 3m
```

Comment:
```suggestion
    golangci-lint run --no-config --disable-all --enable gosimple --enable gofmt --enable errcheck --enable errorlint --enable govet --enable unused --enable goconst --enable bodyclose  --enable misspell --enable nilerr --enable revive --enable unconvert --timeout 3m
```

Reply:
revive seems to give a lot of errors (51 issues found, and some of them are not useful, but will rather make the code more confusing.
Eg.: var varName string = "" ; or var varName bool = false;   <---Remove the = "" or = false, since it's the default value. This makes the code more confusing imo

Spellcheck is useful to run locally before commiting. But it also flags mqtt mosquitto, so we may have to remove it from the script. But it was very useful

Reply:
If that's the case, let's not address those more complex linters in this PR, but make a separate ticket for them. Those will require probably require config files, and maybe even integration with IDE. @AryanGodara Please do not forget to open a ticket for this and once you do, feel free to resolve this conversation and mark the PR as ready.

Reply:
#1774 new issue opened here. Resolving this conversation